dspace-server-webapp/src/test/java/org/dspace/app/rest/ResearcherProfileRestRepositoryIT.java

```diff
@@ -0,0 +1,1049 @@
+    }
+
+    /**
+     * Verify that an user can change the visibility of his profile using the patch
+     * endpoint even if was created by an admin.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testPatchToChangeVisibilityOfProfileCreatedByAnAdmin() throws Exception {
+
+        String id = user.getID().toString();
+
+        String adminToken = getAuthToken(admin.getEmail(), password);
+        String userToken = getAuthToken(user.getEmail(), password);
+
+        getClient(adminToken).perform(post("/api/eperson/profiles/")
+            .param("eperson", id)
+            .contentType(MediaType.APPLICATION_JSON_VALUE))
+            .andExpect(status().isCreated());
+
+        getClient(adminToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isOk());
+
+        // change the visibility to true
+        List<Operation> operations = asList(new ReplaceOperation("/visible", true));
+
+        getClient(userToken).perform(patch("/api/eperson/profiles/{id}", id)
+            .content(getPatchContent(operations))
+            .contentType(MediaType.APPLICATION_JSON_VALUE))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.visible", is(true)));
+
+        getClient(userToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.visible", is(true)));
+    }
+
+    @Test
+    public void testPatchToChangeVisibleAttributeOfNotExistProfile() throws Exception {
+
+        String id = user.getID().toString();
+        String authToken = getAuthToken(user.getEmail(), password);
+
+        getClient(authToken).perform(post("/api/eperson/profiles/")
+                                         .contentType(MediaType.APPLICATION_JSON_VALUE))
+                            .andExpect(status().isCreated())
+                            .andExpect(jsonPath("$.visible", is(false)));
+
+        getClient(authToken).perform(delete("/api/eperson/profiles/{id}", id))
+                            .andExpect(status().isNoContent());
+
+        List<Operation> operations = asList(new ReplaceOperation("/visible", true));
+
+        getClient(authToken).perform(patch("/api/eperson/profiles/{id}", id)
+                                         .content(getPatchContent(operations))
+                                         .contentType(MediaType.APPLICATION_JSON_VALUE))
+                            .andExpect(status().isNotFound());
+    }
+
+    /**
+     * Verify that after an user login an automatic claim between the logged eperson
+     * and possible profiles without eperson is done.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testAutomaticProfileClaimByEmail() throws Exception {
+
+        String id = user.getID().toString();
+
+        String adminToken = getAuthToken(admin.getEmail(), password);
+
+        // create and delete a profile
```

Comment:
This process of first creating & deleting a profile using the REST API is unrealistic for this scenario, as it's very unlikely for a profile to be autoclaimed in this way.  It's much more likely that an Entity of type Person is created that happens to have a matching email address.   

So, I'd recommend we change this to create an Entity of type Person (using the Entity endpoints) and add the user's email address to it.  Then we can prove that Entity will be auto-claimed on next login.

If you'd rather keep this method as-is, then I'd recommend we have a different method test the process I'm describing.  I think we need a test that proves that a regular Entity can be turned into / auto-claimed as a Profile

Reply:
I will create a new test